packages/contracts/README.md

```diff
@@ -27,3 +27,31 @@ interface IFinancialProtocol {
 ```
 
 This protocol allows a service provider to know how long an application has enough funds to run. Of course it is a choice of the service provider off-chain node manager to decide which smart contract implementation to watch for `IFinancialProtocol` events.
+
+## Implementation
+
+The two protocols described above are implemented by two concrete Smart Contracts: `ReaderNodeProvider` and `ValidatorNodeProvider`.
+
+![Alt text](image.png)
+
+### ReaderNodeProvider
```

Comment:
I would format these titles as code, since they are contract names, but it's a purely cosmetic decision.